File: shared_resources/src/main/java/com/thewizrd/shared_resources/helpers/RingerChoice.java
package com.thewizrd.shared_resources.helpers;

import android.util.SparseArray;

public enum RingerChoice {
    SILENT(0),
    VIBRATION(1),
    SOUND(2);

    private final int value;

    public int getValue() {
        return value;
    }

    private RingerChoice(int value) {
        this.value = value;
    }

    private static SparseArray<RingerChoice> map = new SparseArray<>();

    static {
        for (RingerChoice choice : values()) {
            map.put(choice.value, choice);
        }
    }

    public static RingerChoice valueOf(int value) {
        return map.get(value);
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/actions/RingerChoice.kt
package com.thewizrd.shared_resources.actions

import android.util.SparseArray

enum class RingerChoice(val value: Int) {
    SILENT(0),
    VIBRATION(1),
    SOUND(2);

    companion object {
        private val map = SparseArray<RingerChoice>()

        fun valueOf(value: Int): RingerChoice {
            return map[value]
        }

        init {
            for (choice in values()) {
                map.put(choice.value, choice)
            }
        }
    }
}